### PR TITLE
[amg] Azure managed grafana options for notification channels

### PR DESCRIPTION
**Pull request by michelletaal-shell**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
<az grafana>

### General Guidelines

- [ x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
amg

**Comment by yugangw-msft** [comment]
@kairu-ms, can you please review this PR and merge appropriately?

File: src/amg/azext_amg/commands.py [hunk 1]
```diff
@@ -32,6 +32,15 @@ def load_command_table(self, _):
         g.custom_command('query', 'query_data_source')
         g.custom_command('update', 'update_data_source')
 
+    with self.command_group('grafana notification-channel') as g:
+        g.custom_command('list', 'list_notification_channels')
+        g.custom_command('list-short', 'list_notification_channels_short')
```
Comment (yugangw-msft) [inline src/amg/azext_amg/commands.py:37]: I suggest we merge this with `list` command and expose an argument of `--short`

File: src/amg/azext_amg/_params.py [hunk 2]
```diff
@@ -43,6 +43,10 @@ def load_arguments(self, _):
         c.argument("data_source", help="name, id, uid which can identify a data source. CLI will search in the order of name, id, and uid, till finds a match")
         c.argument("definition", help="json string with data source definition, or a path to a file with such content")
 
+    with self.argument_context("grafana notification-channel") as c:
+        c.argument("notification_channel", help="id, uid which can identify a data source. CLI will search in the order of id, and uid, till finds a match")
+        c.argument("definition", help="json string with notification channel definition, or a path to a file with such content")
```
Comment (kairu-ms) [inline src/amg/azext_amg/_params.py:48]: You can use `type=validate_file_or_dict` for json string or json file. https://github.com/Azure/azure-cli-extensions/blob/81dd55b46e58078b172200825be81c1acc44f52f/src/datafactory/azext_datafactory/generated/_params.py#L395-L400

Comment (michelletaal-shell) [reply src/amg/azext_amg/_params.py:48]: Currently, none of the commands have type validation in place. Would you like me to expand my scope slightly and add type validation to all pre-existing commands as well?

Comment (kairu-ms) [reply src/amg/azext_amg/_params.py:48]: Yes, please. BTW, we usually don't need to specify `type` for most simple arguments.

File: src/amg/azext_amg/commands.py (hunk 1 above)
Comment (michelletaal-shell) [reply src/amg/azext_amg/commands.py]: Valid suggestion. Applied in commit [REV1]

File: src/amg/azext_amg/_params.py (hunk 2 above)
Comment (michelletaal-shell) [reply src/amg/azext_amg/_params.py]: Applied in commits [REV2]

Type validation of the argument was added, leading to improved error messaging but also to conflict with the _try_load_file_content function currently executing type validation. This function was therefore omitted for all cli arguments except 'az grafana dashboard import' as this command takes int, str and json objects.

**Comment by kairu-ms** [comment]
Could you add tests for your changes?

**Comment by yugangw-msft** [comment]
@michelletaal-shell your PR looks solid we would like to merge. Do you need any help on adding the test? Assume you have followed the [dev doc](https://github.com/Azure/azure-cli/blob/master/doc/extensions/authoring.md), you can add a test in `/tests/latest/test_amg_scenario.py`, then run `azdev test amg`  which should generate a new test recording yaml file under `/tests/latest/recordings/`. You can then add both files to the PR.

Or I can push the test change to your branch if you prefer.

**Comment by michelletaal-shell** [comment]
> @michelletaal-shell your PR looks solid we would like to merge. Do you need any help on adding the test? Assume you have followed the [dev doc](https://github.com/Azure/azure-cli/blob/master/doc/extensions/authoring.md), you can add a test in `/tests/latest/test_amg_scenario.py`, then run `azdev test amg` which should generate a new test recording yaml file under `/tests/latest/recordings/`. You can then add both files to the PR.
> 
> Or I can push the test change to your branch if you prefer.

Thank you for accepting my pr.

I've been working on writing some tests, but have been otherwise occupied the last few days. I've been having issues with the tests as automatically setting up the Managed Identity requires Azure permissions I don't in my professional subscription. I'll be trying to work my way around that later this week. I suspect this is why there are also no tests for the other API-call functionalities. Once I get over that hump, the actual tests should be in place rather quickly.

**Comment by yugangw-msft** [comment]
@michelletaal-shell, sounds good. If the permission is hard to workaround, `az grafana create` has `skip` arguments to bypass those. Hope it helps.

**Comment by yugangw-msft** [comment]
@michelletaal-shell, let you know that we are on the way to move to Grafana 9.0 with unified alerting. This will light up the contact-point which is similar with notification channel. Let us hold on to this PR and I will work with you in a couple of weeks to migrate this PR to support `contact-point`. Will that work for you?

**Comment by michelletaal-shell** [comment]
@yugangw-msft, thank you for the update. I'm nearly done with setting up tests for all features and will commit those this week. I'm definitely willing to work with you to support contact point. Just let me know when you guys are ready to move forward.

**Comment by kairu-ms** [comment]
@yugangw-msft does this PR ready for review and merge?

**Comment by yugangw-msft** [comment]
@kairu-ms, please merge. 
Note, I will have one more PR for AMG's GA feature to submit this week. Once that is in, I will bump the version to release these 2 change together.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**